src/api/app/models/notifications/rss_feed_item.rb

```diff
@@ -0,0 +1,23 @@
+class Notifications::RssFeedItem < Notifications::Base
+  MAX_ITEMS_PER_USER = 10
+  MAX_ITEMS_PER_GROUP = 10
+
+  def self.cleanup
+    User.all_without_nobody.find_in_batches batch_size: 500 do |batch|
+      batch.each do |user|
+        if user.is_active?
+          ids = user.rss_feed_items.pluck(:id).slice(MAX_ITEMS_PER_USER..-1)
```

Comment:
Not sure if it gains much performance, but you could use an offset here. That might requires to reorder them before.